### DISCUSSION TITLE
System.InvalidOperationException: Invalid State - when using ManagedScanner from version >3.0.0-alpha-0216

### DISCUSSION THREAD

**Discussion by rgomez90** [shinyorg/shiny#1076] [Q&A]
@aritchie I was experimenting a bit with the library, BLE and MAUI (v. 3.0.0-alpha) and I managed to create a `ManagedScanner` and scan some devices when using 3.0.0-alpha-0216 using this basic code (using an android device with API LEVEL 28)

```csharp
public partial class MainPage : ContentPage
    {
        private readonly IBleManager _manager;
        private readonly IManagedScan _scan;

        public MainPage(IBleManager manager)
        {
            _manager = manager;
            _scan = _manager.CreateManagedScanner();
            _scan.Peripherals.CollectionChanged += Peripherals_CollectionChanged;
            InitializeComponent();
        }

        public ObservableCollection<ManagedScanResult> Peripherals => _scan.Peripherals;

        private async void OnButtonClicked(object sender, EventArgs e)
        {
            await _scan.Toggle();
        }

        private void Peripherals_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
        {
            foreach (var result in e.NewItems.Cast<ManagedScanResult>())
            {
                Debug.WriteLine(result.Name);
            }            
        }
    }
```

However when updating the library to latest version 3.0.0-alpha-0266, when pressing the button I got directly (without showing dialog for permissions) a `System.InvalidOperationException: Invalid State NotSetup` raised from `access.Assert()` in `ManagedScan.Start()`.

3.0.0-alpha-0245 shows the same behavior.

3.0.0-alpha-0225 will show the permissions dialog but after granting permissions a `System.InvalidOperationException: Invalid State Denied` is raised.

Downgrading to 3.0.0-alpha-0216 will solve the problem and the scan works as expected.

Since v3.0.0 is alpha I just wanted to ask if this is expected behavior (and need to configure something else in order to make it work with recent releases) or this may be a bug?

**Comment by aritchie** [comment]
I am not offering support for alphas.  If you want to use them, you need to look through the code yourself.  

I will answer this one specific question - the code is pretty much the same as the stable 2.x.  If you are getting that exception, it means you are missing permissions in your Android manifest

**Reply by rgomez90** [reply]
Thank you @aritchie. I completely understand that.

I was indeed yesterday exploring the code and if I am not wrong when using the `ManagedScan` the permissions will be automatically added via `IBleManager.RequestAcccess()` which is called in  `IManagedScan.Start()`, so I supposed there is no need to manually add any permissions to the Android manifest.

What I think is indeed a bug is that `ManagedScan.Toogle()` always return `false` (i.e. `IsScanning` is always `false`), even if the scan works properly.

**Reply by aritchie** [reply]
I do not add values automatically to the manifest in MAUI. 

I'm taking bugs for the stable version only right now because the alpha is NOT DONE.  Please please please follow the guidelines & code of conduct for the repo - https://github.com/shinyorg/shiny/discussions/600
